### PR TITLE
Add mechanism to get blockhash queue length

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
It can be crucial for clients to know if a particular blockhash has expired. `getFeeCalculatorForBlockhash` offers one way to test this, but it would also be helpful for clients to be able to estimate how many slots until expiration. `getRecentBlockhash` returns the slot context of the response, so clients just need the blockhash queue length to track its expiration.

We may also want to be able to change how long a blockhash is valid for, most likely at epoch boundaries.

re: https://github.com/solana-labs/solana/pull/10054#discussion_r425800500

#### Summary of Changes
- Add `getBlockhashLifespan` rpc endpoint. Response includes the relevant epoch, on the assumption that any queue-length changes will be valid for the entire epoch.
- Plumb `getBlockhashLifespan` through client
- Brush up `solana fees`, and add blockhash-lifespan info to it.
- Also sneak in a few camelCase tags that were missing in cli_output

### CODE REVIEW

**Comment by mvines** [comment]
`docs/src/apps/jsonrpc-api.md` 💔

**Comment by mvines** [comment]
Funny thing, I'm currently improving `solana ping` and this is a thing I could immediately use

File: docs/src/apps/jsonrpc-api.md [hunk 1]
```diff
@@ -213,6 +214,32 @@ curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","id":1, "m
...
+
+None
+
+#### Results:
+
+The result will be an RpcResponse JSON object with `value` equal to and object
+with the following fields:
+
+* `epoch: <u64>` - epoch
+* `blockhashQueueLength: <usize>` - length of the blockhash queue
```
Comment (mvines) [inline docs/src/apps/jsonrpc-api.md:231]: There could be more guidance here on what assumptions can be made based off the value of `blockhashQueueLength`.    I think it's: you can assume your blockhash is valid for N slots, where N = "slot that the blockhash you have is from" + blockhashQueueLength.    Is that right?

Comment (mvines) [reply docs/src/apps/jsonrpc-api.md:231]: If so, I wonder if `blockhashQueueLength` would be better named something like `blockhashTTL` or `blockhashLifespan`

Comment (CriesofCarrots) [reply docs/src/apps/jsonrpc-api.md:231]: > I think it's: you can assume your blockhash is valid for N slots, where N = "slot that the blockhash you have is from" + blockhashQueueLength. Is that right?

Yes, exactly!
In the interest of not adding initializations (even though I know its a standard one), I like `blockhashLifespan`.

**Comment by garious** [comment]
What about returning the lifespan via `get_recent_blockhash_with_commitment()`? And better, what about returning a "valid_through_slot_height" instead of a lifespan.  That'd save me the trouble of the next question, where do I get the "birth date" of the blockhash with this "lifespan"?

**Pull request closed without merging**